### PR TITLE
[CP-2165] I want to switch to another device

### PR DESCRIPTION
**Pull request by patryk-sierzega**
JIRA Reference: [CP-2165]

### :memo: Description ️

-

### :muscle: Checklist before requesting a review - nice to have

- getState function in async thunk actions is correctly typed
- redux selectors are used in components / prop drilling is reduce

### :exclamation: Checklist before merging a pull request

- change went through the QA process
- translations are updated in dedicated application
- [CHANGELOG.md](./CHANGELOG.md) is updated

[CP-2165]: https://appnroll.atlassian.net/browse/CP-2165?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by dkarski** [comment]
🟡 
Sure, please pay attention to the sizes of elements and the areas of empty margins, etc., as there is a difference between design and implementation

Zrzut ekranu 2024-01-19 o 09 00 39
Zrzut ekranu 2024-01-19 o 09 00 46

left: feature branch | right: design

Zrzut ekranu 2024-01-19 o 08 58 22
Zrzut ekranu 2024-01-19 o 08 59 13

File: libs/core/__deprecated__/renderer/components/rest/menu/menu-bottom.component.tsx
Comment (dkarski) [inline libs/core/__deprecated__/renderer/components/rest/menu/menu-bottom.component.tsx:85]: 🟢 
I'm considering moving the `SelectDeviceButton` and `ChangeDeviceButton` components to separate files. This would allow for:

1. **Modularity:** Having each component in a separate file would facilitate code organization and development.
2. **Clearer Project Structure:** Separate files would make navigation and understanding of the component hierarchy in the application easier.

File: libs/core/__deprecated__/renderer/locales/default/en-US.json
Comment (dkarski) [inline libs/core/__deprecated__/renderer/locales/default/en-US.json:202]: thanks ❤️ x2

File: libs/core/__deprecated__/renderer/locales/default/en-US.json
Comment (dkarski) [inline libs/core/__deprecated__/renderer/locales/default/en-US.json:77]: thanks ❤️

File: libs/core/device-manager/reducers/device-manager.reducer.ts
Comment (dkarski) [inline libs/core/device-manager/reducers/device-manager.reducer.ts:110]: ⚪ Neutral
maybe toggler? less code

```
toggleSelectDeviceDrawer(true)
toggleSelectDeviceDrawer(false)
```

File: libs/core/device-select/components/drawer-device.component.tsx
Comment (dkarski) [inline libs/core/device-select/components/drawer-device.component.tsx:28]: 🟡 

Your code was missing animations for the hover. Introduced the transition property for smoother animations. Now, changes in background color occur gradually and more smoothly when hovering.

```suggestion
  margin: 1rem 0.5rem 1rem 0.5rem;
  transition: background-color ${transitionTime("veryQuick")} ${transitionTimingFunction("smooth")};
```

File: libs/core/device-select/constants/event.constant.ts
Comment (dkarski) [inline libs/core/device-select/constants/event.constant.ts:9]: ⚪ Neutral

What are your thoughts on using this algorithm for Redux actions?
```
DEVICE_SELECT-CLOSE_SELECT_DEVICE_DRAWER
```

DEVICE_SELECT - Event ID
CLOSE_SELECT_DEVICE_DRAWER? - Redux Action Name

**Review by dkarski — COMMENTED** [review]
I have no objections to the code itself. I will approve after incorporating design-related corrections.

File: libs/core/device-select/constants/event.constant.ts
Comment (patryk-sierzega) [reply libs/core/device-select/constants/event.constant.ts:9]: I am fully convinced, I think I would prefer lowercase :)

File: libs/core/__deprecated__/renderer/components/rest/menu/menu-bottom.component.tsx
Comment (patryk-sierzega) [reply libs/core/__deprecated__/renderer/components/rest/menu/menu-bottom.component.tsx:85]: Not sure about that, SelectDeviceButton and ChangeDeviceButton are only implementations of DeviceButton. I am just passing props to that component, nothing more.

I've moved it out of return to be more readable.

File: libs/core/device-manager/reducers/device-manager.reducer.ts
Comment (patryk-sierzega) [reply libs/core/device-manager/reducers/device-manager.reducer.ts:110]: I think u are suggesting method to simply update state, not a toggler. Toggles changes value to the opposite.

File: libs/core/device-select/components/drawer-device.component.tsx
Comment (patryk-sierzega) [reply libs/core/device-select/components/drawer-device.component.tsx]: I am not missing that, there is no mentioning this in Figma

File: libs/core/device-manager/reducers/device-manager.reducer.ts
Comment (dkarski) [reply libs/core/device-manager/reducers/device-manager.reducer.ts]: Yes, you understood the essence of my suggestion. However, remember that you don't have to consider it at all, as the change is great.

**Review by dkarski — APPROVED** [review]
🟢 
I would suggest removing the cursor pointer change on the active state, as it feels more natural. Instead, let's add a hover animation for devices. This is a minor adjustment, and it's better to do it right away without a separate pull request and an additional ticket. Just need a confirmation from Renata on the Slack channel.

File: libs/core/device-select/components/drawer-device.component.tsx
Comment (dkarski) [reply libs/core/device-select/components/drawer-device.component.tsx]: https://github.com/mudita/mudita-center/pull/1666#pullrequestreview-1836090683

File: libs/core/__deprecated__/renderer/components/rest/menu/menu-bottom.component.tsx
Comment (dkarski) [reply libs/core/__deprecated__/renderer/components/rest/menu/menu-bottom.component.tsx:85]: Moving this outside the main component block could be a compromise for me. However, as I mentioned at the beginning of the comment, please don't get blocked by my remark, as it is not blocking at all.

Comment (dkarski) [reply libs/core/__deprecated__/renderer/components/rest/menu/menu-bottom.component.tsx:85]: Moving this outside the main component block could be a compromise for me. However, as I mentioned at the beginning of the comment, please don't get blocked by my remark, as it is not blocking at all.

**Review by dkarski — DISMISSED** [review]

File: libs/core/__deprecated__/renderer/components/rest/menu/menu-bottom.component.tsx
Comment (dkarski) [inline libs/core/__deprecated__/renderer/components/rest/menu/menu-bottom.component.tsx:121]: please remove this part of the code if you would like to avoid situation like in the bellow screen 👇 
Zrzut ekranu 2024-01-30 o 08 55 33

**Review by dkarski — DISMISSED** [review]
🟢 

I've reviewed your code, and the current method using polling is OK, but it might be worth considering the use of IPC in Electron for greater efficiency. IPC would allow for immediate response to changes without the need for constant state polling.

**Review by dkarski — DISMISSED** [review]

**Review by dkarski — DISMISSED** [review]

**Review by dkarski — APPROVED** [review]

**Pull request merged**
